### PR TITLE
CHAT-523 URLs are not considered anymore as links and enter key does …

### PR DESCRIPTION
**Pull request by trongtt**
…not work properly

### CODE REVIEW

File: application/src/main/webapp/js/chat.js [hunk 1]
```diff
@@ -1442,7 +1442,8 @@ ChatApplication.prototype.initMention = function() {
               $( this ).replaceWith('@' + $( this ).attr('data-mention') + ' ');
             });
 
-            msg = $div.text();
+            // Replace a special character which presents &nbsp; with a normal space.
+            msg = $div.text().replace(new RegExp(String.fromCharCode(160),"g")," ");
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: We could create the RegExp only once and reuse it.

File: application/src/main/webapp/js/chat.js [hunk 2]
```diff
@@ -1442,7 +1442,8 @@ ChatApplication.prototype.initMention = function() {
               $( this ).replaceWith('@' + $( this ).attr('data-mention') + ' ');
             });
 
-            msg = $div.text();
+            // Replace a special character which presents &nbsp; with a normal space.
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: Do we know who adds this character ? Can't we prevent this ?

Comment (trongtt) [inline application/src/main/webapp/js/chat.js]: This is from usage of contenteditable div for Chat input (like wysiwyg ckeditor input). It uses HTML entities for spaces when composing, and we can not prevent it.

File: application/src/main/webapp/js/chat.js (hunk 1 above)
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: ok. Can you add that in the comment please ?

File: application/src/main/webapp/js/chat.js (hunk 2 above)
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: https://github.com/exo-addons/chat-application/pull/75

Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: ok. Can you add this in the comment please ?

Comment (trongtt) [inline application/src/main/webapp/js/chat.js]: Updated

**Comment by trongtt** [comment]
Merged

**Pull request closed without merging**